src/Data/Biapplicative.hs

```diff
@@ -84,6 +90,129 @@ biliftA3 :: Biapplicative w => (a -> b -> c -> d) -> (e -> f -> g -> h) -> w a e
 biliftA3 f g a b c = bimap f g <<$>> a <<*>> b <<*>> c
 {-# INLINE biliftA3 #-}
 
+-- | Traverse a 'Traversable' container in a 'Biapplicative'.
+traverseBia :: (Traversable t, Biapplicative p)
+            => (a -> p b c) -> t a -> p (t b) (t c)
+traverseBia = traverseBiaWith traverse
+-- We use a staged INLINABLE so we can rewrite traverseBia to specialized
+-- versions for a few important types.
+{-# INLINABLE [1] traverseBia #-}
+
+-- | Perform all the 'Biappicative' actions in a 'Traversable' container
+-- and produce a container with all the results.
+--
+-- @
+-- sequenceBia = 'traverseBia" id
+-- @
+sequenceBia :: (Traversable t, Biapplicative p)
+            => t (p b c) -> p (t b) (t c)
+sequenceBia = traverseBia id
+{-# INLINABLE sequenceBia #-}
+
+-- | A version of 'traverseBia' that doesn't care how the traversal is
+-- done.
+--
+-- @
+-- 'traverseBia' = traverseBiaWith traverse
+-- @
```

Comment:
Add laws to explain what this does! They should mirror the `Traversable` laws in some fashion, but presumably use `(,)` instead of `Identity` and one of the bifunctor composition types instead of `Compose`.